lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -1262,51 +1051,55 @@ public function getDateSubYearsExpression($date, $years)
      *
      * @param string $date     The column or literal representing a date to perform the arithmetic operation on.
      * @param string $operator The arithmetic operator (+ or -).
-     * @param int    $interval The interval that shall be calculated into the date.
+     * @param string $interval The interval that shall be calculated into the date.
      * @param string $unit     The unit of the interval that shall be calculated into the date.
-     *                         One of the DATE_INTERVAL_UNIT_* constants.
-     *
-     * @return string
+     *                                 One of the DATE_INTERVAL_UNIT_* constants.
      *
      * @throws DBALException If not supported on this platform.
      */
-    protected function getDateArithmeticIntervalExpression($date, $operator, $interval, $unit)
+    protected function getDateArithmeticIntervalExpression(string $date, string $operator, string $interval, string $unit) : string
```

Comment:
`$interval` should be `int` according to phpDoc so this is a BC break.

Reply:
Addressed in #3498.